master/internal/db/postgres.go

```diff
@@ -1571,6 +1571,38 @@ WHERE id = :id`, setClause(toUpdate)), validation)
 		return errors.Wrapf(err, "error updating (%v) in validation (%v, %v)",
 			strings.Join(toUpdate, ", "), trialID, stepID)
 	}
+
+	if validation.State == model.CompletedState {
```

Comment:
This is the only part I'm not really in love with, those sort by best validation is a very common action and scanning the trials table and the validations table for each trial to order them all is probably ridiculously expensive, so somewhere like this is where we're at.